projects/step-frontend/src/lib/modules/execution/components/execution-progress/execution-progress.component.ts

```diff
@@ -297,6 +303,65 @@ export class ExecutionProgressComponent
         updateSelection: this.isFirstUpdate ? UpdateSelection.ALL : UpdateSelection.ONLY_NEW,
       });
       this.isFirstUpdate = false;
+
+      /**
+       * The following objects have to be defined in order to have a working "dashlet-chart".
+       * The dashlet chart is a wrapper over chart component. The big difference is the API in the fact that the dashlet
+       * item will fetch its data automatically, while the chart component waits for the data to be fed with.
+       *
+       * Context - this is the brain of a dashboard. every event goes through it, and it is responsible for syncing all the
+       * charts, filters, etc
+       *
+       * Dashlet item - these are the settings of a chart. Many settings and outputs does not make sense when using one
+       * single chart only (since it expects to be part of a dashboard with multiple dashlets).
+       * A few would be: shifting/delete outputs, edit mode, inheritGlobal... flags.
+       *
+       *
+       * Now that I see it implemented 'alone', ideally we would make a few improvements to be more easy and natural to use
+       * the chart as a 'standalone' item.
+       * A new component would be ideal, which would be a wrapper over the timeseries-chart core component. We could make
+       * this independent of any context, and react only via simple inputs. The component would look like this:
+       *
+       * <step-standalone-chart
+       *  [range]="{start: execution.startTime, end: execution.endTime}"
+       *  [metricKey]="response-time" -- the current dashlet will have the attributes coming from db, while on this we need to fetch the metric first
+       *  [grouping]="[name]"
+       *  [filters]="...."
+       *  [aggregation]="AVG"
+       *  [unit]="ms"
+       *  [displayType]="line"
+       *  [renderingSettings]="..."
+       * />
+       *
+       */
+
+      this.chartContext = new TimeSeriesContext({
+        id: '123',
```

Comment:
what exactly does this ID reference? Does it make sense to manually reference something in the analytics view? What would happen if we use an ID multiple times?

Reply:
@neogucky the Id is an internal identifier which is mostly used to identify a context (a dashboard). If you have 3 tabs with 3 executions open, you will have 3 contexts created. they will have the executionId as `id`. The id is also used to sync charts (or at least it was).